drivers/overlay/encryption.go

```diff
@@ -224,7 +225,9 @@ func programSA(localIP, remoteIP net.IP, spi *spi, k *key, dir int, add bool) (f
 	if add {
 		action = "Adding"
 		xfrmProgram = ns.NlHandle().XfrmStateAdd
-		crypt = &netlink.XfrmStateAlgo{Name: "cbc(aes)", Key: k.value}
+		salt := make([]byte, 4)
+		binary.LittleEndian.PutUint32(salt, k.tag)
+		crypt = &netlink.XfrmStateAlgo{Name: "rfc4106(gcm(aes))", Key: append(k.value, salt...), ICVLen: 64}
```

Comment:
Regarding the 
> For a given key, the IV MUST NOT repeat. The most natural way to implement this is with a counter

I need to double check what the kernel does if we do not pass anything, but the following statement

```
Note that the encrypter can use any IV generation method that meets the uniqueness requirement, without  coordinating with the decrypter.
```

gives me hope the encrypter and decrypter can autonomously chose a random value, so I am guessing the kernel would do that.

I will confirm, thanks.